text/0000-crates.io-default-ranking.md

```diff
@@ -0,0 +1,1443 @@
+  - Would need to unpack and run this on each package version in a background
+    job started by a publish; then save the percentage in crates.io's database.
+
+- In the crate root documentation, presence of a section headed with the word
+  "Example" and containing a codeblock
+  - Existing issue, seen in the survey results is that people look in both the
+    README of the repo and the front page of the docs for examples. We have an
+    opportunity to encourage at least one to be present reliably.
+  - Increases the doc percentage score by 5%
+
+- Presence of files in `/examples`
+  - Future improvement: [render and link to examples in documentation][examples]
+  - Increases the doc percentage score by 5%
+
+[cargo doc-coverage]: https://crates.io/crates/cargo-doc-coverage
+[examples]: https://github.com/rust-lang/cargo/issues/2760
+
+<table>
+    <tr>
+        <th>Crate</th>
+
+        <th>Doc coverage of top-level items</th>
+        <th>Example in the crate root docs bonus</th>
+        <th>`/examples` bonus</th>
+        <th>Overall Ease of Use score</th>
+    </tr>
+    <tr>
+        <td>peresil</td>
+        <td>10/10, 100%</td>
+        <td>5%</td>
+        <td>0%</td>
+        <td>105%, ☀️</td>
+    </tr>
+    <tr>
+        <td>combine</td>
+        <td>43/44, 98%</td>
+        <td>5%</td>
+        <td>0%</td>
+        <td>103%, ☀️</td>
+    </tr>
+    <tr>
+        <td>nom</td>
+        <td>170/195, 87%</td>
+        <td>5%</td>
+        <td>0%</td>
+        <td>92%, ☀️</td>
+    </tr>
+    <tr>
+        <td>lalrpop</td>
+        <td>4/5, 80%</td>
+        <td>0%</td>
+        <td>0%</td>
+        <td>80%, 🌤</td>
+    </tr>
+    <tr>
+        <td>peg</td>
+        <td>2/3, 66%</td>
+        <td>0%</td>
+        <td>0%</td>
+        <td>66%, 🌥</td>
+    </tr>
+</table>
+
+### Maintenance
+
+- Last released version date: newer is better. This information is already
+  available in crates.io's database; could be stored in the database and
+  updated per-publish. Combined as follows, then reported as a percentage
+  relative to the most released crate.
+  - Number of releases in the last year - 10%
+  - Number of releases in the last 6 mo - 30%
+  - Number of releases in the last month - 60%
+  - Yanked versions are not counted.
+
+- Stable version number
+  - >= 1.0.0 ranks higher than < 1.0.0
+  - >= 1.0.0 increases the maintenance score by 5%.
+
+- Number of owners: more is better.
+  - A GitHub group owner would count as 1.
+  - Future improvement: count # of people in the github group at version
+    publish time
+  - >= 3 owners increases the maintenance score by 5%.
+
+
+We don't have the overall most actively released crate to compute a relative
+release score, so for this analysis we're using the one out of these five
+crates that has the most release activity, peg.
+
+<table>
+    <tr>
+        <th>Crate</th>
+        <th>Releases in last year</th>
+        <th>Releases in last 6 mo</th>
+        <th>Releases in last 1 mo</th>
```

Comment:
It would also mean that projects that release on the same cycle as rust itself (6 weeks) would see this score fluctuate wildly.